src/printer/smt2/smt2_printer.cpp

```diff
@@ -509,7 +509,30 @@ void Smt2Printer::toStream(std::ostream& out,
   // uf theory
   case kind::APPLY_UF: typeChildren = true; break;
   // higher-order
-  case kind::HO_APPLY: break;
+  case kind::HO_APPLY:
+    // collapse "@" chains, i.e.
+    //
+    // @(@(a, b), c) --> (@ a b c)
```

Comment:
This isn't the actual smt2 syntax right? Can you comment on what is actually being printed?

Reply:
the syntax for the above example would be 

`((a b) c)` becomes `(a b c)`

Reply:
I mean, comment in the code :)

Reply:
ah, right hahaha will do